src/App.tsx

```diff
@@ -1,14 +1,39 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
 import React from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  movies: Movie[],
+  query: string,
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    movies: moviesFromServer,
+    query: '',
+  };
+
+  handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      query: event.currentTarget.value,
+    });
+  };
+
+  moviesToShow = () => {
+    const moviesCopy = [...this.state.movies];
```

Comment:
you do a filter, you don't need to make a copy
```suggestion
    const { movies } = this.state;
```